### ISSUE TITLE
[Bug] Uncaught TypeError: Object(...) is not a function when deploying with Azure Pipelines

### ISSUE THREAD

**Issue by AverageCakeSlice** [Azure/static-web-apps#545]
**Description**
I am writing an app using Vue 3 for the SPA and .NET Core 3.1 for the API.

As of yesterday evening, Azure Pipelines seems to be breaking my releases to US Central. It reports that my builds ran successfully, but when I attempt to access the site the app fails to load, with a console error being thrown stating "Uncaught TypeError: Object(...) is not a function" that points to a seemingly arbitrary div element in my code.

This seems to be affecting all of my apps. I have two separate static web apps for dev and test, that track the dev and test branches in my Azure Repo. I have pipelines setup for both that deploy when changes are pushed to their respective branches. I first noticed the issue on Dev, and attempted reverting the changes in git, but the issue persisted. I then made a single comment in test and pushed, which resulted in the same issue, despite the fact that test had none of the changes in dev. Reverting to the previous commit and pushing the changes does not resolve the issue.

This has completely halted development, I don't want to push to our production repository as I'm afraid the issue will cascade down to that application as well.

** Azure Pipelines YAML**
```yml
trigger:
- dev

pool:
  vmImage: ubuntu-latest

steps:
  - checkout: self
    submodules: true
  - task: AzureStaticWebApp@0
    inputs:
      app_location: '/'
      api_location: 'api'
      output_location: 'dist'
      azure_static_web_apps_api_token: $(deployment_token)
```
```yml
trigger:
- test

pool:
  vmImage: ubuntu-latest

steps:
  - checkout: self
    submodules: true
  - task: AzureStaticWebApp@0
    inputs:
      app_location: '/'
      api_location: 'api'
      output_location: 'dist'
      azure_static_web_apps_api_token: $(deployment_token)
```

**Expected behavior**
I expect the build pipeline to deploy and operate identically to how it runs locally with the Azure SWA CLI, barring auth and environment variable differences.

**Screenshots**
**Screenshot of the console error**
![image](https://user-images.githubusercontent.com/37710170/128877473-6635c253-f6a2-421c-84d8-d0da85f33c8d.png)

**Screenshot of the error source**
![image](https://user-images.githubusercontent.com/37710170/128877538-961f8ee7-16af-4295-beab-3a1b17d462f4.png)

**Device info (if applicable):**
 - OS: MacOS 11.5
 - Browser Chrome, Safari

**Additional context**
I am able to run the application locally from my machine using the static web apps CLI (`swa start http://localhost:8080 --api http://localhost:7071`) with `npm run serve` and `func start` for my webapp and API, respectively. I receive no console errors or warnings, and the application launches normally.

**Comment by miwebst** [comment]
Hmm, that's not good! It seems like a build issue to me. When you build the project locally, how does the resulting section look compared to the code deployed to SWA? Trying to understand why building it on SWA leads to different static content.

**Comment by AverageCakeSlice** [comment]
Building the application locally results in a different set of files, and the contents are similar, but the aliases are different

A bit of a follow-up since I last discovered this: For some reason, the application will occasionally switch the file referred to in the error output to one of the bundled files 
![image](https://user-images.githubusercontent.com/37710170/129216437-e973ed11-e1cc-43a5-8148-ec1a9240096d.png)

![image](https://user-images.githubusercontent.com/37710170/129216508-613496f5-403e-4c5e-8745-c169fe72e294.png)

In my local build, the content is aliased to something else on the same line
![image](https://user-images.githubusercontent.com/37710170/129216864-90408f70-42f6-4b0c-a53b-33ff05c886b5.png)

I'm not sure if it's an issue with webpack, or something else. The fact that the issue is also occurring on my test branch without any other changes except a comment leads me to believe that something in the build pipeline may have changed.

@miwebst I noticed that you were working on a PR for the SWA Azure Pipelines Task a few days ago, the timeline matches up with when I first encountered this issue, do you think that there's any relation between this issue and the SWA API changes?

**Comment by AverageCakeSlice** [comment]
Another follow up to this @miwebst, when I build the project locally and run using the npm `http-server` tool, I can run it through the SWA CLI and it works just fine without any errors.

**Comment by AverageCakeSlice** [comment]
I've finally found the solution. After scrolling through the build logs I discovered what I believe to be the source of the issue:
![image](https://user-images.githubusercontent.com/37710170/129377799-fc0ff181-89c7-44e1-a051-1602c0883b70.png)

It turns out I had not checked-in package-lock.json into source control, so I think the build process was automatically pulling in updated dependencies that broke my build when I deployed it, whereas locally I had package-lock.json present, so it was working fine there. 

After adding package-lock.json to source control, the build completed successfully and actually works!

Apologies if this is a silly issue, as I'm relatively new to frontend development. Thank you so much for reaching out @miwebst, and hopefully this can help someone else out who's running into the same issue!
